### PR TITLE
feat: support keystone-auth

### PR DESCRIPTION
**Pull request by okozachenko1203**
fix https://github.com/vexxhost/magnum-cluster-api/issues/9

### CODE REVIEW

**Comment by okozachenko1203** [comment]
This has been tested on local lab

**Review by mnaser — CHANGES_REQUESTED** [review]
I think we can take this as an opportunity to start cleaning up all of our manifests which are bundled.  I think it's very difficult to manage them at the moment and I would like to suggest the following:

1. Create a `charts` folder and start bundling the charts in there, if there is no chart, we build it (in this case for this)
2. Since we already have Helm on the system, we can simply use `helm template` with a custom values file instead of using Jinja2
3. With that in place, we're able to simply just bump the chart data in here and easily change the overrides by relying on `values` instead of raw manifest files.
4. The templated value is what we "feed" into the ClusterResourceSet, and it also makes it easier for the long run to transition to the addon providers as well since we will have the charts ahead of time.

Can you make changes to this to accomodate for that?

**Comment by mnaser** [comment]
@okozachenko1203: this assumes that `magnum-conductor` can reach the cluster directly.  In Atmosphere, it can. However, when users are using OSA or Kolla, that is likely not the case, so we'll have to rely on CAPO for the cluster to be reachable (especially in scenarios when the cluster proxy is running).

File: magnum_cluster_api/resources.py [hunk 1]
```diff
@@ -331,6 +331,65 @@ def get_object(self) -> pykube.ConfigMap:
...
+                        ),
+                        values={
+                            "conf": {
+                                "auth_url": auth_url
+                                + ("" if auth_url.endswith("/v3") else "/v3"),
+                                "ca_file": (
+                                    "/etc/kubernetes/cloud_ca.crt"
+                                    if utils.get_cloud_ca_cert()
+                                    else ""
+                                ),
```
Comment (mnaser) [inline magnum_cluster_api/resources.py:359]: refactor into helm chart value

File: magnum_cluster_api/resources.py [hunk 2]
```diff
@@ -331,6 +331,65 @@ def get_object(self) -> pykube.ConfigMap:
...
+                                                "path": "/etc/kubernetes/cloud_ca.crt",
+                                                "type": "File",
+                                            },
+                                            "name": "cloud-ca-cert-volume",
+                                        }
+                                    ]
+                                    if utils.get_cloud_ca_cert()
+                                    else []
+                                ),
+                            },
```
Comment (mnaser) [inline magnum_cluster_api/resources.py:387]: move logic to helm chart

File: magnum_cluster_api/utils.py [hunk 3]
```diff
@@ -395,3 +396,19 @@ def convert_to_rfc1123(input: str) -> str:
     :rtype: str
     """
     return re.sub(r"[^a-zA-Z0-9]+", "-", input).lower()
+
+
+def get_keystone_auth_default_policy(cluster: magnum_objects.Cluster):
```
Comment (mnaser) [inline magnum_cluster_api/utils.py:401]: lets make this return a dicft

**Review by mnaser — CHANGES_REQUESTED** [review]

**Review by mnaser — APPROVED** [review]

**Pull request merged**